redbox-core/redbox/transform.py

```diff
@@ -233,55 +244,18 @@ def sort_documents(documents: list[Document]) -> list[Document]:
     3.8, foo.txt, 24
     """
 
-    def is_consecutive(a: Document, b: Document) -> bool:
-        """True if two documents have consecutive indices."""
-        within_one = abs(a.metadata["index"] - b.metadata["index"]) <= 1
-        return a.metadata["uri"] == b.metadata["uri"] and within_one
-
     def max_score(group: list[Document]) -> float:
         """Returns the maximum score in a group of documents."""
         return max(d.metadata["score"] for d in group)
 
-    def process_group(group: list[Document]) -> list[list[Document]]:
-        """Breaks a group into blocks of ordered consecutive indices.
-
-        The group is intended to be a single file_name.
-        """
-        # Process consecutive blocks and sort them by index
-        consecutive_blocks = []
-        temp_block = [group[0]]
-
-        for doc in group[1:]:
-            if is_consecutive(temp_block[-1], doc):
-                temp_block.append(doc)
-            else:
-                # Append the current block
-                consecutive_blocks.append(temp_block)
-                temp_block = [doc]
-
-        # Append the last block
-        consecutive_blocks.append(temp_block)
-
-        # Sort each block by index
-        sorted_blocks = [sorted(block, key=lambda d: d.metadata["index"]) for block in consecutive_blocks]
-
-        return sorted_blocks
-
-    # Step 1: Sort by file_name and then index to prepare for grouping consecutive documents
-    documents_sorted = sorted(documents, key=lambda d: (d.metadata["uri"], d.metadata["index"]))
-
-    # Step 2: Group by file_name and handle consecutive indices
-    grouped_by_file = itertools.groupby(documents_sorted, key=lambda d: d.metadata["uri"])
+    # Group and sort docs by file_name and handle consecutive indices
+    grouped_by_file = group_documents(documents)
 
-    # Process each group
-    all_sorted_blocks = []
-    for _, group in grouped_by_file:
-        group = list(group)  # Convert the iterator to a list
-        sorted_blocks = process_group(group)
-        all_sorted_blocks.extend(sorted_blocks)
+    # Step 1: Process each group
+    all_sorted_blocks = sum((group_and_sort_documents(list(group)) for _, group in grouped_by_file), [])
```

Comment:
I've got to be honest, I can't make any sense of this line. What is the sum over? What are we 'processing'?